src/Symfony/Component/Cache/Adapter/CouchbaseBucketAdapter.php

```diff
@@ -0,0 +1,226 @@
+        string $bucket,
+        string $namespace,
+        int $defaultLifetime,
+        array $options = [],
+        ?MarshallerInterface $marshaller = null
+    ): void {
+        if (!static::isSupported()) {
+            throw new CacheException('Couchbase >= 2.6.0 is required');
+        }
+
+        $this->maxIdLength = static::MAX_KEY_LENGTH;
+
+        $this->bucket = $client->openBucket($bucket);
+
+        $this->bucket->operationTimeout = $options['operationTimeout'] ?? $this->bucket->operationTimeout;
+        $this->bucket->configTimeout = $options['configTimeout'] ?? $this->bucket->configTimeout;
+        $this->bucket->configNodeTimeout = $options['configNodeTimeout'] ?? $this->bucket->configNodeTimeout;
+
+        parent::__construct($namespace, $defaultLifetime);
+        $this->enableVersioning();
+        $this->marshaller = $marshaller ?? new DefaultMarshaller();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function doFetch(array $ids)
+    {
+        $resultsCouchbase = $this->bucket->get($ids);
+
+        $results = [];
+        foreach ($resultsCouchbase as $key => $value) {
+            if (null !== $value->error) {
+                continue;
+            }
+            $results[$key] = $this->marshaller->unmarshall($value->value);
+        }
+
+        return $results;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function doHave($id): bool
+    {
+        return false !== $this->bucket->get($id);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function doClear($namespace): bool
+    {
+        if ('' === $namespace) {
+            $this->bucket->manager()->flush();
+
+            return true;
+        }
+
+        return false;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function doDelete(array $ids): bool
+    {
+        $results = $this->bucket->remove(array_values($ids));
+
+        foreach ($results as $key => $result) {
+            if (null !== $result->error && static::KEY_NOT_FOUND !== $result->error->getCode()) {
+                continue;
+            }
+            unset($results[$key]);
+        }
+
+        return 0 === \count($results);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function doSave(array $values, $lifetime)
+    {
+        if (!$values = $this->marshaller->marshall($values, $failed)) {
+            return $failed;
+        }
+
+        $lifetime = $this->normalizeExpiry($lifetime);
+
+        $ko = [];
+        foreach ($values as $key => $value) {
+            $result = $this->bucket->upsert($key, $value, ['expiry' => $lifetime]);
+
+            if (null !== $result->error) {
+                $ko[$key] = $result;
+            }
+        }
+
+        return [] === $ko ? true : $ko;
+    }
+
+    /**
+     * @param int $expiry
+     *
+     * @return int
+     */
+    private function normalizeExpiry(int $expiry): int
+    {
+        if ($expiry && $expiry > static::THIRTY_DAYS_IN_SECONDS) {
```

Comment:
why that? same as memcache maybe: couchbase considers big enough numbers as timestamp instead of lifetimes?

Reply:
The expiration time of the Couchbase document is used in seconds. If it is greater than 30 days (606024 * 30), the server will interpret it as absolute UNIX time (seconds since the period 1970-01-01T00: 00: 00). The same that memcached